### DISCUSSION TITLE
What is Wix 4 built on top of?

### DISCUSSION THREAD

**Discussion by devray90** [wixtoolset/issues#7661] [Questions]
Hi,

Could someone give me an overview of what Wix 4 and its relative components are built on top of so i know what i need to install pre req wherever I need to run my msi?

Are they .net framework based or .net core and what version? I noticed custom actions are still .net framework based, but the bootstrapper is .net 6 compatible? Its a bit confusing, could someone clarify?

Thanks

**Comment by barnson** [comment]
WiX doesn't impose any pre-reqs. Code you write might, so if you write a managed BA using .NET 6, you'll have a pre-req or use self-contained.

**Reply by devray90** [reply]
Ok, what about specifically managed code and custom actions? From the looks of it on Wix 4 they are still framework based? Is there a plan to move them to .net 6 eventually?

**Reply by robmen** [reply]
All custom actions provided in the WiX Toolset are written in native code and work on all Windows versions supported by Microsoft.
